pkg/webhook/controlplane/ensurer.go

```diff
@@ -521,21 +517,41 @@ func (e *ensurer) addAdditionalFilesForResolvConfOptions(options []string, new *
 		permissions int32 = 0755
 		template          = `#!/bin/sh
 
-file=/run/systemd/resolve/resolv.conf
-tmp=/etc/resolv.conf.new
-dest=/etc/resolv.conf
+tmp=/etc/resolv-for-kubelet.conf.new
+dest=/etc/resolv-for-kubelet.conf
 line=%q
 
-if [ -f "$file" ]; then
-  cp "$file" "$tmp"
-  echo "" >> "$tmp"
-  echo "# updated by update-resolv-conf.service (installed by gardener-extension-provider-openstack)" >> "$tmp"
-  echo "$line" >> "$tmp"
-  mv "$tmp" "$dest"
-fi`
+is_systemd_resolved_system()
+{
+    if [ -f /run/systemd/resolve/resolv.conf ]; then
+      return 0
+    else
+      return 1
+    fi
+}
+
+rm -f "$tmp"
+if is_systemd_resolved_system; then
+  if [ "$line" = "" ]; then
+    ln -s /run/systemd/resolve/resolv.conf "$tmp"
+  else
+    cp /run/systemd/resolve/resolv.conf "$tmp"
```

Comment:
We could copy `/run/systemd/resolve/resolv.conf` always to `$tmp` and only append `$line` if not empty. wdyt?
But it's just a nit.

Reply:
I think it is easier to reason about the configuration with the symbolic link. Otherwise you have to compare the file content.

Reply:
Hmm I don't get this, but as said this was just a nit. I'm also fine with this approach.